### ISSUE TITLE
Relation to STI has wrong 'mappedBy' reference type

### ISSUE THREAD

**Issue by milo526** [mikro-orm/mikro-orm#5850]
### Describe the bug

Defining a relation to the base entity of a STI entity yields a Metadata Error

> MetadataError: "Entity.relation" has wrong 'mappedBy' reference type: "top most class" instead of "entity"

### Reproduction

https://codesandbox.io/p/devbox/zealous-chebyshev-d5nymr?file=%2Fsrc%2Fexample.test.ts%3A21%2C1

### What driver are you using?

@mikro-orm/sqlite

### MikroORM version

next

### Node.js version

Node: v20.12.0, Typescript: v5.5.3

### Operating system

Code sandbox

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by milo526** [comment]
Issue seems to be in the `@mikro-orm/core` package and seems like it's not driver specific.

Issue remains when NOT using `abstract` top most class.

**Comment by B4nan** [comment]
The reproduction seems to be working if you add the `Readable` entity to the config explicitly. It should be auto-discovered, but it is apparently not, and it results in the child entities being discovered as normal ones, producing wrong schema in the first place.

```diff
-entities: [User, Book, Magazine],
+entities: [User, Book, Magazine, Readable],
```

**Comment by B4nan** [comment]
fixed via 7c1976f0d03fcef2012fbf3e87158cae4cfee81a
